runbotics-orchestrator/src/main/java/com/runbotics/service/impl/ProcessCollectionServiceImpl.java

```diff
@@ -38,16 +40,45 @@ public ProcessCollectionServiceImpl(
         this.userService = userService;
     }
 
-    public List<ProcessCollectionDTO> getCollectionsByCriteria(ProcessCollectionCriteria criteria) {
-        if (criteria.getParentId() != null) {
+    public void checkCollectionAvailability(UUID collectionId, User user) {
+        if (processCollectionRepository.countCollectionById(collectionId) == 0) {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Cannot find process collection");
+        }
+
+        boolean hasUserAccessEveryCollection = user.getFeatureKeys().contains(FeatureKeyConstants.PROCESS_COLLECTION_EVERY);
+        if (!hasUserAccessEveryCollection) {
+            boolean isCollectionAvailable = processCollectionRepository.countAvailableCollectionById(collectionId, user) != 0;
```

Comment:
it's tricky because if processCollectionRepository.countAvailableCollectionById(collectionId, user) will return null, some obj. etc. isCollectionAvailable will be set to true
try
 boolean isCollectionAvailable = !(processCollectionRepository.countAvailableCollectionById(collectionId, user) == 0);